benchmarks/osb/extensions/param_sources.py

```diff
@@ -35,29 +61,143 @@ def __init__(self, workload, params, **kwargs):
         self.percent_completed = 0
         self.offset = 0
 
-    def _read_data_set(self):
-        if self.data_set_format == HDF5DataSet.FORMAT_NAME:
-            return HDF5DataSet(self.data_set_path, Context.INDEX)
-        if self.data_set_format == BigANNVectorDataSet.FORMAT_NAME:
-            return BigANNVectorDataSet(self.data_set_path)
+    def _read_data_set(self, data_set_format: str, data_set_path: str,
+                       data_set_context: Context):
+        if data_set_format == HDF5DataSet.FORMAT_NAME:
+            return HDF5DataSet(data_set_path, data_set_context)
+        if data_set_format == BigANNVectorDataSet.FORMAT_NAME:
+            return BigANNVectorDataSet(data_set_path)
         raise ConfigurationError("Invalid data set format")
 
     def partition(self, partition_index, total_partitions):
-        if self.data_set.size() % total_partitions != 0:
-            raise ValueError("Data set must be divisible by number of clients")
+        """
+        Splits up the parameters source so that multiple clients can read data
+        from it.
+        Args:
+            partition_index: index of one particular partition
+            total_partitions: total number of partitions data set is split into
+
+        Returns:
+            The parameter source for this particular partion
+        """
+        if self.num_vectors % total_partitions != 0:
```

Comment:
is this mean that the data set size must be divisible by the number of parallel clients? 
If so I think in next revision we need to relax this requirement and divide evenly except for last client that will have the remainder

Reply:
Yes thats a good point. I can update this in a future PR. Will create an issue when PR is merged.